packages/kernel/src/kernel.ts

```diff
@@ -533,14 +533,18 @@ export abstract class BaseKernel implements IKernel {
   private async _execute(msg: KernelMessage.IMessage): Promise<void> {
     const executeMsg = msg as KernelMessage.IExecuteRequestMsg;
     const content = executeMsg.content;
-    this._executionCount++;
+    if (content.store_history) {
```

Comment:
[Upstream](https://github.com/jupyter/jupyter_client/blob/main/jupyter_client/client.py#L559-L565), it looks like we may want to:

- check if `content.store_history !== false` (as it should be true by default)
- check if `content.silent` is true, which would override it

Reply:
I had assumed that the documented behavior was enforced somewhere, but testing it it does indeed seem to not be the case (i.e., running with just `silent: true` does not lead to `store_history: false`). Would be simple to include here, but are you suggesting that this really needs to be fixed in another component (like the `jupyter_client`)? I don't know enough about how all of the pieces of the jupyter framework work together to know where to actually fix this (took me long enough to figure out that the execution counter was increased here!).

Reply:
> documented behavior 

all depends on which one you read, really, and some of it is _only_ in the reference implementation. 

>  fixed in another component 

for purposes of jupyterlite, the only component we get to reuse is [ipython](https://github.com/ipython/ipython/search?l=Python&q=execution_count): we have to do everything else ourselves in this repo, as `ipykernel` and `jupyter_client` do all of the zmq stuff (which doesn't work in the browser). so really, this behavior is probably _already_ covered in ipython, and we just need to wire it up... except it's on the "wrong" side of the kernel wall. Putting it in our base `kernel` implementation is fine, though, if it means other kernel authors don't have to think about it for it to work "right" for different use cases.

>  don't know enough about how all of the pieces of the jupyter framework

No worries, there's rather a lot to keep straight when we're shipping client/server/kernel in the same repo! in the Bad Old Days, this is what ipython was like... we're just getting to put all the shiny pieces back together... even here, we want to get to a multi-repo setup (e.g. #386), but for now this is what we have.

ideally, we'd figure out some way to wire up [jupyter_kernel_test](https://github.com/jupyter/jupyter_kernel_test) and write some conformance tests against our kernels: for now, just pointing to how our stuff quacks like the upstream is sufficient and helpful: liberal links to (versioned) places in the reference implementations would be great... it all gets minimized anyway.

Reply:
> we just need to wire it up... except it's on the "wrong" side of the kernel wall. Putting it in our base kernel implementation is fine, though, if it means other kernel authors don't have to think about it for it to work "right" for different use cases.

Yeah I also thought we were tracking the execution count on Python side for the pyolite kernel.

Probably this is fine for now to add the check in the base kernel class for now since this is where the tracking is done. We can open an issue to track this.

For reference JupyterLab default to `store_history: true` here: https://github.com/jupyterlab/jupyterlab/blob/c5df985383964b027e475e667d69a7e8facf42bb/packages/services/src/kernel/default.ts#L710